lib/bibliothecary/dependency.rb

```diff
@@ -0,0 +1,39 @@
+module Bibliothecary
+  # @attr_reader [String] name The name of the package, e.g. "ansi-string-colors"
+  # @attr_reader [String] requirement The version requirement of the release, e.g. "1.0.0" or "^1.0.0"
+  # @attr_reader [String] type The type of dependency, e.g. "runtime" or "test"
+  # @attr_reader [Boolean] direct Is this dependency a direct dependency (vs transitive dependency)? (default: nil)
+  # @attr_reader [Boolean] deprecated Is this dependency deprecated? (default: nil)
```

Comment:
Q: opinions on defaulting these to something, instead of `nil`?

Reply:
Were you thinking some explicit `unknown` value, or making an assumption (eg default to direct = true)? For `type` and `direct` I'm not sure there's an obvious default to me. `deprecated = false` could make sense though.

Introducing that kind of change makes this less of drop-in replacement, but I suppose this could be an opportune time to return something with less ambiguity?

Reply:
good points, we could always return later to set defaults. I'll just keep the `deprecated` and `optional` defaults of `false` in here since they seem less contentious, and we could think about the others later.

Reply:
actually wait, those are both fields specific to certain platforms (Golang, Maven, etc), so maybe I should keep them `nil` defaults for now.